src/main/java/org/apache/maven/buildcache/CacheControllerImpl.java

```diff
@@ -166,22 +166,22 @@ public CacheResult findCachedBuild( MavenSession session, MavenProject project,
         ProjectsInputInfo inputInfo = projectInputCalculator.calculateInput( project );
 
         final CacheContext context = new CacheContext( project, inputInfo, session );
-        // remote build first
-        CacheResult result = findCachedBuild( mojoExecutions, context );
```

Comment:
@gnodet  @maximilian-novikov-db remote honestly this change brings regression.
There are 2 local caches: 
1) local cache of remote repository
2) cache of local builds

Previously, if remote build was present in local cache it was always used - there was no lookup. Now local build used with priority which could lead to difficult to debug issues.

Reply:
Well, that was exactly what I was trying to fix.  If there was no remote build cached, but a local build, a lookup did occur...

Reply:
> @AlexanderAshitkin Well, that was exactly what I was trying to fix. If there was no remote build cached, but a local build, a lookup did occur... This basically means that local builds are useless if you have a remote cache, because they are never used, which is a bit weird.

Not exactly. This is lookup for a local build (which is not from remote cache)
```
        // local build first
        CacheResult result = findLocalBuild( mojoExecutions, context );
```
This is lookup to find cached build from remote repo (if missed - then lookup remote cache):
```
    CacheResult remoteBuild = findCachedBuild( mojoExecutions, context );
```
Old logic was:
1) Lookup local cache of remote repo
2) If #1 failed lookup remote repo and download
3) If #2 failed (effectively no remote build) - rebuild locally

Now logic is:
1) Try to use local build even if build is present in remote repo
2) Try to find locally cached remote build
3) Loockup remote cache

The new scheme has minimal performance benefits (both tried to find local build first), but likely will lead to less consistent results and difficult to debug discrepancies in presence of remote cache